contracts/strategies/SnowballStrategyV1.sol

```diff
@@ -0,0 +1,235 @@
+            _burn(msg.sender, amount);
+            totalDeposits = totalDeposits.sub(depositTokenAmount);
+            emit Withdraw(msg.sender, depositTokenAmount);
+        }
+    }
+
+    function _withdrawDepositTokens(uint amount) private {
+        require(amount > 0, "SnowballStrategyV1::_withdrawDepositTokens");
+        uint sharesAmount = amount.mul(1e18).div(snowGlobe.getRatio());
+        stakingContract.withdraw(sharesAmount);
+        snowGlobe.withdraw(sharesAmount);
+    }
+
+    function reinvest() external override onlyEOA {
+        uint unclaimedRewards = checkReward();
+        require(unclaimedRewards >= MIN_TOKENS_TO_REINVEST, "SnowballStrategyV1::reinvest");
+        _reinvest(_convertRewardIntoWAVAX(unclaimedRewards));
+    }
+
+    function _convertRewardIntoWAVAX(uint pendingReward) private returns (uint) {
+        stakingContract.getReward();
+        DexLibrary.swap(
+            rewardToken.balanceOf(address(this)),
+            address(rewardToken), address(WAVAX),
+            swapPairWAVAXSnob
+        );
+        return WAVAX.balanceOf(address(this));
+    }
+
+    /**
+     * @notice Reinvest rewards from staking contract to deposit tokens
+     * @dev Reverts if the expected amount of tokens are not returned from `stakingContract`
+     * @param amount deposit tokens to reinvest
+     */
+    function _reinvest(uint amount) private {
+        uint devFee = amount.mul(DEV_FEE_BIPS).div(BIPS_DIVISOR);
+        if (devFee > 0) {
+            _safeTransfer(address(WAVAX), devAddr, devFee);
+        }
+
+        uint adminFee = amount.mul(ADMIN_FEE_BIPS).div(BIPS_DIVISOR);
+        if (adminFee > 0) {
+            _safeTransfer(address(WAVAX), owner(), adminFee);
+        }
+
+        uint reinvestFee = amount.mul(REINVEST_REWARD_BIPS).div(BIPS_DIVISOR);
+        if (reinvestFee > 0) {
+            _safeTransfer(address(WAVAX), msg.sender, reinvestFee);
+        }
+
+        uint depositTokenAmount = DexLibrary.convertRewardTokensToDepositTokens(
+            amount.sub(devFee).sub(adminFee).sub(reinvestFee),
+            address(WAVAX),
+            address(depositToken),
+            swapPairToken0,
+            swapPairToken1
+        );
+
+        _stakeDepositTokens(depositTokenAmount);
+        totalDeposits = totalDeposits.add(depositTokenAmount);
+
+        emit Reinvest(totalDeposits, totalSupply);
+    }
+    
+    /**
+     * @notice Safely transfer using an anonymosu ERC20 token
+     * @dev Requires token to return true on transfer
+     * @param token address
+     * @param to recipient address
+     * @param value amount
+     */
+    function _safeTransfer(address token, address to, uint256 value) private {
+        require(IERC20(token).transfer(to, value), 'SnowballStrategyV1::TRANSFER_FROM_FAILED');
+    }
+
+    function checkReward() public override view returns (uint) {
+        return DexLibrary.estimateConversionThroughPair(
+            stakingContract.earned(address(this)),
```

Comment:
could add the current contract balance of the reward token here to handle dust (instead of in the other place)

Reply:
added